### PR TITLE
GAP under PTS testing

### PR DESCRIPTION
**Pull request by rymanluk**
One major and two minor patches which came up during PTS GAP testing

### CODE REVIEW

File: apps/bletiny/src/main.c [hunk 1]
```diff
@@ -936,12 +936,23 @@ bletiny_gap_event(struct ble_gap_event *event, void *arg)
...
+        console_printf("received advertisement; event_type=%d rssi=%d "
+                       "addr_type=%d addr=", event->disc.event_type,
+                       event->disc.rssi, event->disc.addr_type);
         print_addr(event->disc.addr);
-        console_printf(" length_data=%d rssi=%d data=",
-                       event->disc.length_data, event->disc.rssi);
+
+        /*
+         * There is no adv data to print in case of connectable
+         * direct advertising
```
Comment (andrzej-kaczmarek) [inline apps/bletiny/src/main.c]: direct -> directed

File: apps/bletiny/src/main.c [hunk 2]
```diff
@@ -936,12 +936,23 @@ bletiny_gap_event(struct ble_gap_event *event, void *arg)
...
+         */
+        if (event->disc.event_type == BLE_HCI_ADV_TYPE_ADV_DIRECT_IND_HD ||
+                event->disc.event_type == BLE_HCI_ADV_TYPE_ADV_DIRECT_IND_LD) {
+                console_printf("\nConnection directed advertising event\n");
```
Comment (andrzej-kaczmarek) [inline apps/bletiny/src/main.c]: Connection -> connectable

File: apps/bletiny/src/main.c [hunk 3]
```diff
@@ -141,7 +141,24 @@ bletiny_print_adv_fields(const struct ble_hs_adv_fields *fields)
     int i;
 
     if (fields->flags_is_present) {
-        console_printf("    flags=0x%02x\n", fields->flags);
+        console_printf("    flags=0x%02x:\n", fields->flags);
+
+        if (!(fields->flags & BLE_HS_ADV_F_DISC_LTD) &&
+                !(fields->flags & BLE_HS_ADV_F_DISC_GEN)) {
+                console_printf("        Non discoverable mode\n");
```
Comment (andrzej-kaczmarek) [inline apps/bletiny/src/main.c]: Non discoverable -> Non-discoverable

Comment (rymanluk) [inline apps/bletiny/src/main.c]: I can always count on you :) Will fix  all of these

**Comment by rymanluk** [comment]
fixed according to Andrzej comments

**Comment by ccollins476ad** [comment]
Looks good.  Thanks for catching that mutex bug!

**Pull request merged**